generator/mavgen_c.py

```diff
@@ -103,10 +103,6 @@ def generate_main_h(directory, xml):
 
 // MESSAGE LENGTHS AND CRCS
 
-#ifndef MAVLINK_MESSAGE_LENGTHS
-#define MAVLINK_MESSAGE_LENGTHS {${message_lengths_array}}
-#endif
```

Comment:
This needs more work:

- mavlink_helpers from v0.9 and v1.0 weren't changed so they still need this; there are also tests using it
- the comment above saying `// MESSAGE LENGTHS AND CRCS` is now a bit strange when the only definition is then `MAVLINK_MESSAGE_CRCS` (renaming this to `MAVLINK_MESSAGE_ENTRIES` is probably the best idea)
- the `message_lengths_array` variable is unused now, we should remove its generation too

Reply:
> - mavlink_helpers from v0.9 and v1.0 weren't changed so they still need this; there are also tests using it

I think the implementation is correct for v0.9 and v1.0 because only msgids up to 256 were available.

> - the comment above saying // MESSAGE LENGTHS AND CRCS is now a bit strange when the only definition is then MAVLINK_MESSAGE_CRCS (renaming this to MAVLINK_MESSAGE_ENTRIES is probably the best idea)

Right, thanks.

> - the message_lengths_array variable is unused now, we should remove its generation too

I think I checked that and found that it was still used in other places (such as for v1.0).

Reply:
> I think the implementation is correct for v0.9 and v1.0 because only msgids up to 256 were available.

Sorry, I might have explained poorly. You are removing the macro definition, which is used in the code - in v0.9, v1.0 and tests (tests seem to actually be covered by a local definition).

> I think I checked that and found that it was still used in other places (such as for v1.0).

I think that's the confusion you are making. The macro is used but you are removing it, which makes the Python variable useless - I can't find it being used anywhere else but here.

> Right, thanks.

I take my advice back. The `MAVLINK_MESSAGE_CRCS` only contains CRCs for MAVLink1 so renaming it isn't the best idea - but you are either going to revert removing `MAVLINK_MESSAGE_LENGTHS ` or change the MAVLink1 version to also include lengths, so the comment is likely to be fixed one way or the other.

Reply:
Thanks, I'll follow up on that.

Reply:
Restored it in f1ee3622ea59137471b3755cb94449c249cb14c1.